app/Http/Controllers/v1/UserActivityMapController.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Http\Controllers\v1;
+
+use App\Http\Controllers\Controller;
+use App\Models\UserCourseActivity;
+use Illuminate\Support\Facades\DB;
+
+class UserActivityMapController extends Controller
+{
+    public function show($id)
+    {
+        $aggregatedData = UserCourseActivity::where('user_id', $id)
+            ->select(
+                'user_id',
```

Comment:
So i tried for a while to make this work with ORM non-raw Sql. Couldn't get it to work
so this is just a nit (verbosity):
```php
        $aggregatedData = UserCourseActivity::where('user_id', $id)
            ->selectRaw(
                'user_id, DATE(date) as date,
                SUM(external_has_activity) as active_course_count,
                SUM(external_total_activity_time_delta) as total_activity_time'
            )
            ->groupBy('user_id', 'date')
            ->get();
```

Reply:
I took another run at making more ORM, but in the end I left it alone since it functions.